contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
> do you really need to put "not" between the scope and the keyword

Currently yes. I'm not opposed to supporting `"not system:windows"` in addition to the current syntax though.

> and noticed that if there's no scope then scope "configurations" is assumed, should I reflect this in the PR

No, that's a convenience for simple projects. Your current approach is better in this case.

> seems like "iif" isn't documented in the wiki

Quite possible—feel free to add it!